### PR TITLE
in game ui

### PR DESCRIPTION
**Pull request by tommydgithub**
Player one game ui its in duplicate scene. use z,x,c,v key right now to change it.

### CODE REVIEW

File: Assets/Scripts/UIManager.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+
+
+public class UIManager : MonoBehaviour
+{
+    // Start is called before the first frame update
+
```
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:11]: fix weird chunk of space here.

File: Assets/Scripts/UIManager.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+ 
+    public TextMeshProUGUI playerOneScoreLabel;
```
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:14]: 1. All the variables are not matching our naming scheme, please read the c# style guide on the repo's root
2. From the basic principle of encapsulation you should never make variables public unless for special use cases. If you want to access the variable in the unity editor u can add a `[Serializefield]` attribute to the variable, please take reference to other's script.

File: Assets/Scripts/UIManager.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    public TextMeshProUGUI playerOneLivesLeftLabel;
+    public GameObject playerOneDurability;
+    public GameObject playerOneHealth;
+    public GameObject playerOneInventory;
+
+
+    public int playerOneScoreNum;
```
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:21]: You shouldn't make 4 variables for each player. You should make a class that represents a player's UI status and make 4 of that class for each player.

File: Assets/Scripts/UIManager.cs [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    //public int playerTwoScoreNum;
+    //public int playerTwoDurabilityScore;
+    //public int playerTwoHealthScore;
+    //public string playerTwoInventoryType;
+
+
+
+    private void Update()
+    {
+        if (Input.GetKeyDown("z"))
```
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:40]: 1. We are not using the old input system already, so please do not use these codes here.
2. You should hook these functions of updating UI to some events about players getting a score, changing health, etc.

The thing you wrote here is doing nothing in terms of gameplay.

File: Assets/Scripts/UIManager.cs [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+        if (Input.GetKeyDown("v"))
+        {
+            changePlayerOneLivesLeft(-1);
+        }
+
+    }
+
+
+    public void changePlayerOneScore(int changeBy)
```
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:66]: Again, as mentioned, these functions should be in their own class.
Also, the function name is not matching the naming scheme, please fix them all.
Please also take your naming seriously, the name `delta` might be a better naming for the parameter.

File: Assets/Scripts/UIManager.cs [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    {
+        playerOneDurabilityScore = playerOneDurabilityScore + changeBy;
+        playerOneDurability.GetComponent<Image>().fillAmount = playerOneDurabilityScore;
+        //change player
+    }
+
+    public void changePlayerOneHealth(float changeBy)
+    {
+        playerOneHealthScore = playerOneHealthScore + changeBy;
+        playerOneHealth.GetComponent<Image>().fillAmount = playerOneHealthScore;
```
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:83]: Don't use `GetComponent<Image>()` every time the function is called, please make it a variable and set it up from when the player is ready/ when appropriate.

File: Assets/Scripts/UIManager.cs [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+
+
+
+
+
+
+
+
+    //public void changePlayerTwoScore(int changeBy)
```
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:102]: Don't leave chunks of commented code in your PR, your PR should be a finished feature.
The only commented code you could leave is should be a possible fix for edge cases or possibly useful code for new features.

Also from what I've seen the player two status UI is not finished yet. Make sure you've done what your task's saying before thinking of your tasks as finished.

File: Assets/Scripts/UIManager.cs [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+    //    //change player
+    //}
+
+
+    public void changePlayerOneWeaponType(string weaponType)
+    {
+        playerOneInventoryType = weaponType;
+
+        //change player one weapon image
```
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:128]: Weird commenting style. I don't really know why you're commenting like this but from school, we always put commented on top of the code we want to describe. Please fix it.

File: Assets/Scripts/UIManager.cs [hunk 9]
```diff
@@ -0,0 +1,139 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+
+
+public class UIManager : MonoBehaviour
```
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:8]: All the scripts of our game should be under the namespace `Game`

File: Assets/Scripts/UIManager.cs [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+    public GameObject playerOneDurability;
+    public GameObject playerOneHealth;
+    public GameObject playerOneInventory;
+
+
+    public int playerOneScoreNum;
+    public int playerOneLivesLeftNum;
+    public float playerOneDurabilityScore = 0.2f;
+    public float playerOneHealthScore;
+    public string playerOneInventoryType;
```
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:25]: If you are storing different types of "something" please do not use a `string`. You should always make an `enum' for your type. 
Also, I'm not sure what this variable is used for, please briefly explain.

File: Assets/Scripts/UIManager.cs (hunk 9 above)
Comment (JimmyC7834) [inline Assets/Scripts/UIManager.cs:8]: This class should not be called `UIManager`, UI manager should be a script managing or setting up the UIs. You are working on the player's status UI, which this class should be called something like `UI_PlayerStatusDisplay`.

**Review by JimmyC7834 — CHANGES_REQUESTED** [review]

**Comment by JimmyC7834** [comment]
@tommydgithub Please change your code as requested. 
From what I've seen here, your push is not fulfilling what your task promised to do. What we are expecting should be a working UI that updates as players' status changes. Using keyboard input here is doing nothing for our gameplay and should be considered unfinished work.
Also, not being mean, just stating facts. The code I saw is very naive and lacks professionalism, I think reviewing others' code and seeing how we do our work can definitely helps you in your work.
In short, do more code reviews for other PR and fix your code.

**Comment by JimmyC7834** [comment]
Rewroted the whole thing in #40
